undertow/src/main/java/org/wildfly/extension/undertow/HostSingleSignOnDefinition.java

```diff
@@ -22,20 +22,41 @@
 
 package org.wildfly.extension.undertow;
 
+import org.jboss.as.clustering.controller.Capability;
 import org.jboss.as.clustering.controller.ResourceDescriptor;
 import org.jboss.as.clustering.controller.SimpleResourceRegistration;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.capability.RuntimeCapability;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
-
 /**
  * @author Paul Ferraro
  */
 public class HostSingleSignOnDefinition extends SingleSignOnDefinition {
 
+    //we use a runtime API of Object as a hack, so we can check for the presence of the capability in a DUP
+    public static final RuntimeCapability<Object> HOST_SSO_CAPABILITY = RuntimeCapability.Builder.of(Capabilities.CAPABILITY_HOST_SSO, true, new Object())
+            .addRequirements(Capabilities.CAPABILITY_UNDERTOW)
+            .setServiceType(SingleSignOnService.class)
+            .setDynamicNameMapper(pathElements -> new String[]{
+                    pathElements.getParent().getParent().getLastElement().getValue(),
+                    pathElements.getParent().getLastElement().getValue()})
+            .build();
+
     @Override
     public void registerOperations(ManagementResourceRegistration registration) {
         ResourceDescriptor descriptor = new ResourceDescriptor(this.getResourceDescriptionResolver())
-                .addAttributes(SingleSignOnDefinition.Attribute.class)
-                ;
+                .addAttributes(SingleSignOnDefinition.Attribute.class).addCapabilities((Capability) new Capability() {
```

Comment:
FYI - once https://github.com/wildfly/wildfly/pull/10065 is merged, this becomes:

    .addCapabilities(() -> HOST_SSO_CAPABILITY)